app/models/shrimpo.rb

```diff
@@ -0,0 +1,122 @@
+class Shrimpo < ApplicationRecord
+  extend FriendlyId
+  friendly_id :slug_candidates, use: :slugged
+
+  include ActionView::Helpers::DateHelper
+
+  belongs_to :user
+  has_many :shrimpo_entries
+  has_many :posts, as: :postable
+
+  has_one_attached :zip
+  has_one_attached :cover_art
+
+  validates :title, presence: true
+  validates :emoji, presence: true
+  validates :rule_pack, presence: true
+
+  enum status: [:running, :voting, :completed]
+
+  attr_accessor :duration
+
+  after_create :queue_end_shrimpo_job
+
+  VALID_DURATIONS = [
+    # minors
```

Comment:
entries to a minor shrimpo are anonymized until the shrimpo is over

entries to a major shrimpo are public even before the voting starts

Reply:
oh wow ok didn't even think about that but it makes sense to me. how does one choose one type over the other?

Reply:
anything less than 1 month is minor otherwise major

https://github.com/datafruits/datafruits/pull/1153/files#diff-772707db9b2e37e2a455175d81bff546fa82999037cc833679a195aa4d284f06R41